Tools/Pipeline/MainWindow.cs

```diff
@@ -197,7 +199,10 @@ public void UpdateProperties()
 
         public void OutputAppend(string text)
         {
-            Application.Instance.Invoke(() => buildOutput.WriteLine(text));
+            if(Dispatcher.FromThread(Thread.CurrentThread)==null)
+                Application.Instance.AsyncInvoke(() => buildOutput.WriteLine(text));
```

Comment:
I tried to leave the current behavior unchanged just in case...
I think Eto uses Eto.Threading.Thread.IsMain but that throws an exception. 
I'll just call AsuncInvoke for now and wait for @cra0zy 's feedback on this one.